server.go

```diff
@@ -225,6 +251,10 @@ type Server struct {
 	Unsafe bool
 	// If NotifyStartedFunc is set is is called, once the server has started listening.
 	NotifyStartedFunc func()
+	// ReaderBuilder is optional, allows customization of the process that reads DNS frames
+	ReaderBuilder ReaderBuilder
```

Comment:
Don't really like the 'Builder' part of the name, would ReaderFunc be more appropriate? And the comment needs a closing dot.